src/org/pentaho/reporting/platform/plugin/SimpleReportingComponent.java

```diff
@@ -452,6 +453,15 @@ public void setInputs( final Map<String, Object> inputs ) {
     if ( inputs.containsKey( DASHBOARD_MODE ) ) {
       dashboardMode = "true".equals( String.valueOf( inputs.get( DASHBOARD_MODE ) ) ); //$NON-NLS-1$
     }
+    if ( inputs.containsKey( QUERY_LIMIT ) ) {
+      try {
+        if ( this.getReport() != null && inputs.get( QUERY_LIMIT ) != null ) {
+          this.getReport().setQueryLimit( Integer.parseInt( (String) inputs.get( QUERY_LIMIT ) ) );
```

Comment:
A Web UI is by definition always an untrusted source of input. You know that you can change URL parameters with relative ease and that you can even use other tools to send request to a server, right? I dont care what the UI code says, as long as I can send a request via HTTP I can program my own UI that ignores all *your* UI code. And then your limits, if not enforced on the server when the request is processed, are all for nothing.

At no point ever must you accept values from the outside world and NOT check them.